examples/all-clusters-app/p6/src/AppTask.cpp

```diff
@@ -0,0 +1,233 @@
+
+    return err;
+}
+
+void AppTask::AppTaskMain(void * pvParameter)
+{
+    AppEvent event;
+    CHIP_ERROR err = sAppTask.Init();
+    if (err != CHIP_NO_ERROR)
+    {
+        P6_LOG("AppTask.Init() failed");
+        appError(err);
+    }
+
+    P6_LOG("App Task started");
+
+    while (true)
+    {
+        BaseType_t eventReceived = xQueueReceive(sAppEventQueue, &event, pdMS_TO_TICKS(10));
+        while (eventReceived == pdTRUE)
+        {
+            sAppTask.DispatchEvent(&event);
+            eventReceived = xQueueReceive(sAppEventQueue, &event, 0);
+        }
+    }
+}
+
+void AppTask::LightActionEventHandler(AppEvent * aEvent)
+{
+    /* ON/OFF Light Led based on Button interrupt */
+    sLightLED.Invert();
+
+    /* Update OnOff Cluster state */
+    sAppTask.OnOffUpdateClusterState();
+}
+
+void AppTask::ButtonEventHandler(uint8_t btnIdx, uint8_t btnAction)
+{
+    if (btnIdx != APP_LIGHT_BUTTON_IDX)
+    {
+        return;
+    }
+
+    AppEvent button_event              = {};
+    button_event.Type                  = AppEvent::kEventType_Button;
+    button_event.ButtonEvent.ButtonIdx = btnIdx;
+    button_event.ButtonEvent.Action    = btnAction;
+
+    if ((btnIdx == APP_LIGHT_BUTTON_IDX) && (btnAction == APP_BUTTON_RELEASED))
+    {
+        button_event.Handler = LightActionEventHandler;
+        sAppTask.PostEvent(&button_event);
+    }
+}
+
+void AppTask::PostEvent(const AppEvent * aEvent)
+{
+    if (sAppEventQueue != NULL)
+    {
+        BaseType_t status;
+        if (xPortIsInsideInterrupt())
+        {
+            BaseType_t higherPrioTaskWoken = pdFALSE;
+            status                         = xQueueSendFromISR(sAppEventQueue, aEvent, &higherPrioTaskWoken);
+
+#ifdef portYIELD_FROM_ISR
+            portYIELD_FROM_ISR(higherPrioTaskWoken);
+#elif portEND_SWITCHING_ISR // portYIELD_FROM_ISR or portEND_SWITCHING_ISR
+            portEND_SWITCHING_ISR(higherPrioTaskWoken);
+#else                       // portYIELD_FROM_ISR or portEND_SWITCHING_ISR
+#error "Must have portYIELD_FROM_ISR or portEND_SWITCHING_ISR"
+#endif // portYIELD_FROM_ISR or portEND_SWITCHING_ISR
+        }
+        else
+        {
+            status = xQueueSend(sAppEventQueue, aEvent, 1);
+        }
+
+        if (!status)
+            P6_LOG("Failed to post event to app task event queue");
+    }
+    else
+    {
+        P6_LOG("Event Queue is NULL should never happen");
+    }
+}
+
+void AppTask::DispatchEvent(AppEvent * aEvent)
+{
+    if (aEvent->Handler)
+    {
+        aEvent->Handler(aEvent);
+    }
+    else
+    {
+        P6_LOG("Event received with no handler. Dropping event.");
+    }
+}
+
+void AppTask::OnOffUpdateClusterState(void)
+{
+    uint8_t newValue = sLightLED.Get();
+    // write the new on/off value
+    EmberAfStatus status = emberAfWriteAttribute(2, ZCL_ON_OFF_CLUSTER_ID, ZCL_ON_OFF_ATTRIBUTE_ID, CLUSTER_MASK_SERVER, &newValue,
+                                                 ZCL_BOOLEAN_ATTRIBUTE_TYPE);
```

Comment:
Please use
```suggestion
    EmberAfStatus status = chip::app::Clusters:OnOff::Attributes::OnOff::Set(2, newValue);
```
possibly with some `using namespace` above to make this shorter.  Will need to include `app-common/zap-generated/attributes/Accessors.h`

Reply:
@bzbarsky-apple thanks for your review comments.. I have addressed all these comments on new commit